packages/sanity/src/_internal/cli/util/compareStudioDependencyVersions.ts

```diff
@@ -0,0 +1,82 @@
+import path from 'node:path'
+
+import resolveFrom from 'resolve-from'
+import semver from 'semver'
+
+import {type AutoUpdatesImportMap} from './getAutoUpdatesImportMap'
+import {readPackageJson} from './readPackageJson'
+
+async function getRemoteResolvedVersion(fetchFn: typeof fetch, url: string) {
+  try {
+    const res = await fetchFn(url, {method: 'HEAD', redirect: 'manual'})
+    return res.headers.get('x-resolved-version')
+  } catch (err) {
+    throw new Error(`Failed to fetch remote version for ${url}: ${err.message}`)
+  }
+}
+
+interface CompareStudioDependencyVersions {
+  pkg: string
+  installed: string
+  remote: string
+}
+
+/**
+ * Compares the versions of dependencies in the studio with their remote versions.
+ *
+ * This function reads the package.json file in the provided working directory, and compares the versions of the dependencies
+ * specified in the `autoUpdatesImports` parameter with their remote versions. If the versions do not match, the dependency is
+ * added to a list of failed dependencies, which is returned by the function.
+ *
+ * The failed dependencies are anything that does not strictly match the remote version.
+ * This means that if a version is lower or greater by even a patch it will be marked as failed.
+ *
+ * @param AutoUpdatesImportMap - An object mapping package names to their remote import URLs.
```

Comment:
`@params` should reference the parameter name, not the type. The type is inferred.

```suggestion
 * @param autoUpdatesImports - An object mapping package names to their remote import URLs.
```